xla/service/cpu/onednn_matmul_rewriter.cc

```diff
@@ -809,42 +809,46 @@ class OneDnnPostRewriteVisitor : public DfsHloRewriteVisitor {
     }
     auto weights = custom_call->operand(1);
     if (weights->user_count() > 1) {
-      return absl::FailedPreconditionError("Weights could not be prepacked.");
+      return absl::FailedPreconditionError(
+          "Cannot prepack weights. There is more than one consumer.");
     }
     auto weights_shape = weights->shape();
     Literal weights_literal;
-    if (weights_shape.rank() == 2 &&
-        evaluator_.TryEvaluate(weights, &weights_literal)) {
-      auto plain_weights_md = ShapeToMemDesc(weights_shape);
-      if constexpr (std::is_same<PrimDesc,
-                                 dnnl::matmul::primitive_desc>::value) {
-        auto backend_config = custom_call->backend_config<BackendConfig>();
-        if (!backend_config.ok()) {
-          return absl::CancelledError("Weights could not be prepacked.");
-        }
-        if (backend_config->onednn_matmul_config().transpose_b()) {
-          plain_weights_md = TransposeLastTwoDims(plain_weights_md).value();
-        }
-      }
-      TF_RETURN_IF_ERROR(SetWeightsPrepack<PrimDesc>(custom_call, true));
-      auto prim_desc = CreateOneDnnPrimDesc<PrimDesc>(custom_call);
-      auto packed_weights_md = prim_desc->weights_desc();
-      auto packed_weights_shape = MemDescToXlaShapeFlattened(packed_weights_md);
-      auto packed_weights_literal = Literal(packed_weights_shape);
-      ReorderWeight(plain_weights_md, weights_literal.untyped_data(),
-                    packed_weights_md, packed_weights_literal.untyped_data());
-      HloInstruction* reordered_weight = custom_call->AddInstruction(
-          HloInstruction::CreateConstant(std::move(packed_weights_literal)));
-      auto status =
-          custom_call->ReplaceOperandWithDifferentShape(1, reordered_weight);
-      if (!status.ok()) {
-        TF_RETURN_IF_ERROR(SetWeightsPrepack<PrimDesc>(custom_call, false));
-        return absl::CancelledError("Weights could not be prepacked.");
-      } else {
-        return custom_call;
+    if (!(weights_shape.rank() == 2 &&
+          evaluator_.TryEvaluate(weights, &weights_literal))) {
+      return absl::CancelledError(
+          "Cannot prepack weights. Not constant 2D weights.");
+    }
+    auto plain_weights_md = ShapeToMemDesc(weights_shape);
+    if constexpr (std::is_same<PrimDesc, dnnl::matmul::primitive_desc>::value) {
+      auto backend_config = custom_call->backend_config<BackendConfig>();
+      if (!backend_config.ok()) {
+        return absl::CancelledError(
+            "Cannot prepack weights. Customcall does not have a "
+            "BackendConfig.");
```

Comment:
Nit: Just noticed that we can use `TF_ASSIGN_OR_RETURN` here.
```suggestion
      TF_ASSIGN_OR_RETURN(auto backend_config, custom_call->backend_config<BackendConfig>());
```

Reply:
Thanks! changed accordingly.